GitUI/CommandsDialogs/FormClone.cs

```diff
@@ -405,10 +407,16 @@ private void ToSelectedIndexChanged(object sender, EventArgs e)
 
         private readonly AsyncLoader _branchListLoader = new();
 
-        private void UpdateBranches(RemoteActionResult<IReadOnlyList<IGitRef>> branchList)
+        private void UpdateBranches(RemoteActionResult<IReadOnlyList<IGitRef>>? branchList)
         {
             Cursor = Cursors.Default;
 
+            if (branchList == null)
+            {
+                // cancelled
+                return;
+            }
```

Comment:
This pattern occurs in several places. This `UpdateBranches` method is the callback passed to an instance of `AsyncLoader` and the argument is almost never null.

`AsyncLoader` has a behaviour where if its load is cancelled, it invokes the callback with a null argument. A future change should review this, and probably remove that behaviour. It results in lots of these checks, which could be avoided.